src/modules/rlm_files/rlm_files.c

```diff
@@ -38,23 +38,27 @@ typedef struct {
 
 	char const 	*filename;
 	rbtree_t 	*common;
+	fr_trie_t	*common_trie;
```

Comment:
```
typedef struct {
    char const *filename;
    union {
        rbtree_t    *whole;
        fr_trie_t    *prefix;
    } tree;
} rlm_file_t;
```

Don't really care about the type/field names, just the duplication.  There's three reasons to use a union here:

- It shows someone reading the code that the two trees are used exclusively.
- The code will (likely) break instead of continuing silently if we try and access the wrong type of tree.
- It uses slightly less memory (really the least compelling reason here).

You probably need an enum with the tree types here as well unless it's easy to infer which one is in use.